### PR TITLE
Allow json mapgen to use vehicle groups

### PR DESCRIPTION
**Pull request by KiwiTrinsic**
This will allow json mapgens to use vehicle group ids in addition to single vehicle ids. To demonstrate the usefulness of this, I also updated the cars generated in garages to work off a list of common household vehicles instead of statically spawning just a single type.

### CODE REVIEW

**Comment by jokermatt999** [comment]
Nice addition! Always good to have some more random flavor.

**Comment by John-Candlebury** [comment]
Just wondering, will the garage cars spawn all beat up or in pristine condition?

Because well they should probably spawn undamaged almost all the time.

**Comment by KiwiTrinsic** [comment]
John,
I believe they have usually spawned in good condition in the past. I didn't change that part of them, just changed it so they can be different types of cars.

**Comment by DavidKeaton** [comment]
Then that means the condition will be the same as before. Nice.

File: src/mapgen.cpp [hunk 1]
```diff
@@ -754,7 +754,7 @@ class jmapgen_vehicle : public jmapgen_piece {
         if( !x_in_y( chance.get(), 100 ) ) {
             return;
         }
-        m.add_vehicle( type, x, y, rotation[rng(0, rotation.size()-1)], fuel, status );
+        m.add_vehicle( type, point(x, y), rotation[rng(0, rotation.size()-1)], fuel, status );
```
Comment (DavidKeaton) [inline src/mapgen.cpp]: if there is a 3d version of the second argument, use that instead (unless it takes more code, like if the function only takes [x, y] as arguments to begin with).

**Pull request merged**